rcl/test/rcl/test_graph.cpp

```diff
@@ -1354,3 +1388,160 @@ TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_rcl_service_server_
   wait_for_service_state_to_change(false, is_available);
   ASSERT_FALSE(is_available);
 }
+
+/* Test passing invalid params to server_is_available
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_bad_server_available) {
+  // Create a client which will be used to call the function.
+  rcl_client_t client = rcl_get_zero_initialized_client();
+  auto ts = ROSIDL_GET_SRV_TYPE_SUPPORT(test_msgs, srv, BasicTypes);
+  const char * service_name = "/service_test_rcl_service_server_is_available";
+  rcl_client_options_t client_options = rcl_client_get_default_options();
+  rcl_ret_t ret = rcl_client_init(&client, this->node_ptr, ts, service_name, &client_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_ret_t ret = rcl_client_fini(&client, this->node_ptr);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  });
+  // Check, knowing there is no service server (created by us at least).
+  bool is_available;
+  ret = rcl_service_server_is_available(this->node_ptr, &client, &is_available);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  ASSERT_FALSE(is_available);
+
+  ret = rcl_service_server_is_available(nullptr, &client, &is_available);
+  EXPECT_EQ(RCL_RET_NODE_INVALID, ret);
+  rcl_reset_error();
+  rcl_node_t not_init_node = rcl_get_zero_initialized_node();
+  ret = rcl_service_server_is_available(&not_init_node, &client, &is_available);
+  EXPECT_EQ(RCL_RET_NODE_INVALID, ret);
+  rcl_reset_error();
+}
+
+/* Test passing invalid params to get_node_names functions
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_bad_get_node_names) {
+  rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
+
+  rcutils_string_array_t node_names_2 = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t node_namespaces_2 = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t node_enclaves = rcutils_get_zero_initialized_string_array();
+  rcl_ret_t ret = RCL_RET_OK;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    ret = rcutils_string_array_fini(&node_names);
+    EXPECT_EQ(RCUTILS_RET_OK, ret);
+    ret = rcutils_string_array_fini(&node_namespaces);
+    EXPECT_EQ(RCUTILS_RET_OK, ret);
+    ret = rcutils_string_array_fini(&node_enclaves);
+    EXPECT_EQ(RCUTILS_RET_OK, ret);
+  });
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+
+  ret = rcl_get_node_names(nullptr, allocator, &node_names, &node_namespaces);
```

Comment:
@Blast545 here and below, I think it'd be helpful to have a comment explaining what is being tested.